### DISCUSSION TITLE
Risk at assign parameter to i2s

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#758] [Q&A] [answered]
i am sending file to esp1 to esp2 and then at esp2 use decoder . 

I can assign config parameter to i2s via decoder.setNotifyAudioChange(i2s);

But my concern is for example in MP3 or wav file config parameter like sampling frequency  ,  sampling bit lies within approx. 44 bytes of file starts .and as we know espnow data their is chance of data missing so what if the data mis in staring Bytes then how decoder can assign i2s configs ??

if later in file any data mis its not issue but stating data is important ecause of config parameter how to handle this ?

**Comment by pschatzmann** [comment, marked as answer]
If you know the parameters in advance, it is good practice to set I2S up with the expected audio configuration. 
For wav, you would just not play anything which does not have a valid header.
For mp3 I think it does not matter
